### PR TITLE
Add graphQL example

### PR DESCRIPTION
**Pull request by prakhargupta1**
App:

Screenshot 2023-08-18 at 10 07 57 AM

- [x] I've read and followed the [contributing guide](https://github.com/mui/mui-toolpad/blob/master/CONTRIBUTING.md#sending-a-pull-request) on how to create great pull requests.
- [x] I've updated the relevant documentation for any new or updated feature
- [x] I've linked relevant GitHub issue with "Closes #<issue id>"
- [x] I've added a visual demonstration under the form of a screenshot or video

### CODE REVIEW

File: examples/graphQL/toolpad/resources/function.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+export async function queryStargazers(owner: string, repository: string) {
+  if (!process.env.GITHUB_TOKEN) {
+    throw new Error(`Env variable GITHUB_TOKEN not configured`);
+  }
+
+  const endpoint = 'https://api.github.com/graphql';
+  const token = process.env.GITHUB_TOKEN;
+
+  const query = `
```
Comment (apedroferreira) [inline examples/graphQL/toolpad/resources/function.ts:11]: We can use https://www.npmjs.com/package/graphql-tag for these template strings.

**Comment by apedroferreira** [comment]
We should just call the example folder `graphql` I think, as it seems that the pattern is to use all lowercase letters and spacing with hyphens when necessary (as in `npm-stats`)

**Review by apedroferreira — CHANGES_REQUESTED** [review]
Looks good, just really minor comments!
Also, based on the screenshot, we could format the followers data so that we show only the `totalCount` in that column? Unless there's more data we might want to show sometimes.

File: examples/graphQL/toolpad/resources/function.ts (hunk 1 above)
Comment (prakhargupta1) [reply examples/graphQL/toolpad/resources/function.ts]: I added this and expected syntax highlighting, but nothing happened.

**Comment by prakhargupta1** [comment]
> Looks good, just really minor comments/suggestions! Also, based on the screenshot, we could format the followers data so that we show only the `totalCount` in that column? Unless there's more data we might want to show sometimes.

Couldn't make this work, do I need to change the response array?

**Comment by prakhargupta1** [comment]
> We should just call the example folder `graphql` I think, as it seems that the pattern is to use all lowercase letters and spacing with hyphens when necessary (as in `npm-stats`)

Did graphQL -> graphql. This change is visible in my local but isn't detected by GitHub desktop :/ https://github.com/desktop/desktop/issues/2672#issuecomment-1681471505

**Comment by apedroferreira** [comment]
> Did graphQL -> graphql. This change is visible in my local but isn't detected by GitHub desktop :/ [desktop/desktop#2672 (comment)](https://github.com/desktop/desktop/issues/2672#issuecomment-1681471505)

I just added a commit with the change.
It's an issue on Mac/Linux with case sensitive file names it seems.
I ran `git config core.ignorecase true` in the terminal before renaming the folder, and then ran `git config --unset core.ignoreCase` to put it back to normal.

**Comment by prakhargupta1** [comment]
> I just added a commit with the change.
> It's an issue on Mac/Linux with case sensitive file names it seems.
> I used an intermediary commit with `graphqltemp` as the folder name to make it work.

Thanks for this!

**Review by apedroferreira — APPROVED** [review]

**Pull request merged**